ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/json/ToJsonStringTest.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.json;
+
+import static org.junit.Assert.assertEquals;
+
+import io.confluent.ksql.schema.ksql.SqlTimeTypes;
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.junit.Test;
+
+public class ToJsonStringTest {
+  private final ToJsonString udf = new ToJsonString();
+
+  @Test
+  public void shouldSerializeInt() {
+    // When:
+    final String result = udf.toJsonString(123);
+
+    // Then:
+    assertEquals("123", result);
+  }
+
+  @Test
+  public void shouldSerializeBoolean() {
+    // When:
+    final String result = udf.toJsonString(true);
+
+    // Then:
+    assertEquals("true", result);
+  }
+
+  @Test
+  public void shouldSerializeLong() {
+    // When:
+    final String result = udf.toJsonString(123L);
+
+    // Then:
+    assertEquals("123", result);
+  }
+
+  @Test
+  public void shouldSerializeDouble() {
+    // When:
+    final String result = udf.toJsonString(123.456d);
+
+    // Then:
+    assertEquals("123.456", result);
+  }
+
+  @Test
+  public void shouldSerializeDecimal() {
+    // When:
+    final String result = udf.toJsonString(new BigDecimal("0.33333"));
+
+    // Then:
+    assertEquals("0.33333", result);
+  }
+
+  @Test
+  public void shouldSerializeString() {
+    // When:
+    final String result = udf.toJsonString("abc");
+
+    // Then:
+    assertEquals("\"abc\"", result);
+  }
+
+  @Test
+  public void shouldSerializeBytes() {
+    // When:
+    final String result = udf.toJsonString(ByteBuffer.allocate(4).putInt(1097151));
+
+    // Then:
+    assertEquals("\"ABC9vw==\"", result);
+  }
+
+  @Test
+  public void shouldSerializeDate() {
+    // When:
+    final String result = udf.toJsonString(SqlTimeTypes.parseDate("2022-01-20"));
+
+    // Then:
+    assertEquals("\"2022-01-20\"", result);
+  }
+
+  @Test
+  public void shouldSerializeTime() {
+    // When:
+    final String result = udf.toJsonString(SqlTimeTypes.parseTime("06:02:20.588"));
```

Comment:
Ok, out of curiosity, where did the milliseconds get dropped in the process?

Reply:
I made the `to_json_string` UDF use the formatting functions used to output the same types in console. For example, this is the output format for the ksqlDB `TIME` data type:

```sql
ksql> CREATE STREAM time_stream (k STRING KEY, t TIME) WITH (kafka_topic='test', value_format='JSON', partitions=1);

 Message
----------------
 Stream created
----------------
ksql> insert into time_stream (k, t) values ('k1', '06:02:20.588');
ksql> select * from time_stream;
+-------------------------------------------------------------------------------+-------------------------------------------------------------------------------+
|K                                                                              |T                                                                              |
+-------------------------------------------------------------------------------+-------------------------------------------------------------------------------+
|k1                                                                             |06:02:20                                                                       |
^CQuery terminated
```